app/timetables/admin.py

```diff
@@ -8,31 +8,62 @@
 @admin.register(Weekday, MealOption, Course)
 class DefaultAdmin(admin.ModelAdmin):
     """Default admin for models with just name and slug fields."""
+
     readonly_fields = ('slug',)
 
 
+class TimeStampAdmin(admin.ModelAdmin):
+    """Default admin for models having only readonly fields from TimeStampMixin."""
+
+    readonly_fields = ('date_created', 'date_modified')
+
+
 @admin.register(Meal)
 class MealAdmin(DefaultAdmin):
+    """Admin customisation for Meal model."""
+
     fields = ('name', 'slug', 'start_time', 'end_time')
 
 
 class AdminsInline(admin.TabularInline):
+    """Tabular inline setting for Timetable admins."""
+
     model = Timetable.admins.through
 
 
 @admin.register(Timetable)
-class TimetableAdmin(DefaultAdmin):
+class TimetableAdmin(admin.ModelAdmin):
+    """Admin customisation for Timetable model."""
+
+    readonly_fields = ('slug', 'date_created', 'date_modified')
     fields = ('name', 'slug', 'code', 'api_key', 'cycle_length',
               'current_cycle_day', 'description')
```

Comment:
@andela-kanyanwu: You forgot to include 'date_created' and 'date_modified' in fields.